odl/operator/operator.py

```diff
@@ -1402,17 +1715,14 @@ def __init__(self, op, vector):
         self._op = op
         self._vector = vector
 
-    def _call(self, x):
-        """Return ``op(x)``."""
-        # pylint: disable=protected-access
-        return self._op._call(self._vector * x)
-
-    def _apply(self, x, out):
-        """Implement ``op(x, out=out)``."""
-        # pylint: disable=protected-access
-        tmp = self.domain.element()
-        tmp.multiply(self._vector, x)
-        self._op._apply(tmp, out)
+    def _call(self, x, out=None):
+        """Implement ``self(x[, out])``."""
+        if out is None:
+            return self._op(x * self._vector)
```

Comment:
I actually tried to do as you write, mirror the definition. Failed on this one obviously.